### PR TITLE
Prepare release

### LINKED ISSUE #73: Require Ansible 2.9.10+ and add deprecation notices to modules
##### SUMMARY

We should somehow (ideally in some standard way - `galaxy.yml`?) require at least Ansible >=2.9.10 so we can add the correct information to module deprecation notices.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME

All modules in here

##### ADDITIONAL INFORMATION

Currently the sanity tests have to include ignores for `pylint:ansible-deprecated-no-collection-name` - this can be dropped and the necessary info added when we add a requirement for 2.9.10+.

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Do some generic cleanup; set deprecation versions to 2.0.0.

Fixes #73.

##### ISSUE TYPE
- Bugfix Pull Request
- Feature Pull Request

##### COMPONENT NAME
collection

### CODE REVIEW

**Comment by MarkusTeufelberger** [comment]
The ignores in the sanity tests could also be deactivated I think?

**Comment by felixfontein** [comment]
@MarkusTeufelberger did that in ec839dc9b3cb07c2cc182e34c01cf9bf3845e929 (forgot to commit that before the first push...)

**Comment by MarkusTeufelberger** [comment]
Maybe a small "Roadmap/future deprecations" section in the README would be nice to have? On the other hand it is already quite clear that pyopenssl will be gone with 2.0.0 - still it could calm down some people by ensuring them that we aren't going to end support as long as there are upstream supported mainstream distributions (Debian, Ubuntu, CentOS, Fedora, Alpine, Arch, Suse, ...?) out there that wouldn't be able to run these modules with native packages.

**Comment by felixfontein** [comment]
I've updated the README a bit, and fleshed out the roadmap section.

**Comment by felixfontein** [comment]
Finally passes all tests :)

**Comment by felixfontein** [comment]
@MarkusTeufelberger mind if I merge this? Or do you want to check the README.md changes first (or some other changes)?

File: README.md [hunk 1]
```diff
@@ -64,19 +70,27 @@ We're following the general Ansible contributor guidelines; see [Ansible Communi
 If you want to clone this repositority (or a fork of it) to improve it, you can proceed as follows:
 1. Create a directory `ansible_collections/community`;
 2. In there, checkout this repository (or a fork) as `crypto`;
-3. Add the directory containing `ansible_collections` to your [ANSIBLE_COLLECTIONS_PATHS](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#collections-paths).
+3. Add the directory containing `ansible_collections` to your [ANSIBLE_COLLECTIONS_PATH](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#collections-paths).
+
+See [Ansible's dev guide](https://docs.ansible.com/ansible/devel/dev_guide/developing_collections.html#contributing-to-collections) for more information.
```
Comment (MarkusTeufelberger) [inline README.md:75]: Minor nit: Maybe we should not link to the "devel" version in the future here

File: README.md [hunk 2]
```diff
@@ -64,19 +70,27 @@ We're following the general Ansible contributor guidelines; see [Ansible Communi
...
 
 ## Release notes
 
-<!--Add a link to a changelog.md file or an external docsite to cover this information. -->
+See the [changelog](https://github.com/ansible-collections/collection_template/blob/master/CHANGELOG.rst).
```
Comment (MarkusTeufelberger) [inline README.md:79]: dead link

File: README.md (hunk 1 above)
Comment (felixfontein) [reply README.md:75]: True. But the non-devel link does not exist yet (at least that anchor).

File: README.md (hunk 2 above)
Comment (felixfontein) [reply README.md:79]: It will exist once the first release is created :)

Comment (MarkusTeufelberger) [reply README.md:79]: In collection_template?

Comment (felixfontein) [reply README.md:79]: LOL, great catch! I really wonder where I copied that from...

**Comment by MarkusTeufelberger** [comment]
shipit :-)

**Comment by felixfontein** [comment]
@MarkusTeufelberger thanks a lot for reviewing this!

**Pull request merged**